src/main/scala/mesosphere/mesos/TaskGroupBuilder.scala

```diff
@@ -83,6 +86,31 @@ object TaskGroupBuilder extends StrictLogging {
     (executorInfo.build, taskGroup.build, resourceMatch.hostPorts)
   }
 
+  // This method just double-checks that the matched resources are exactly
+  // what is needed and in right quantities.
+  @SuppressWarnings(Array("ComparingFloatingPointTypes"))
+  private[this] def verifyResources(pod: PodDefinition, matchedResources: Seq[mesos.Resource]): Unit = {
+    assert(matchedResources.count(_.getName == "cpus") <= 1)
+    assert(matchedResources.count(_.getName == "mem") <= 1)
+    assert(matchedResources.count(_.getName == "disk") <= 1)
+    assert(matchedResources.count(_.getName == "gpus") <= 1)
```

Comment:
I wonder if we could define resources that it's impossible to have multiple definitions. I'm having a similar issue with the Rust client for Mesos.

Reply:
I don't really know :(

Reply:
In the case of disk resources, it is definitely possible to have multiple matched resources (there are different types of disk resources: root, path, mount)

Reply:
This is not about persistent volumes.

Reply:
I talked to Greg about it, and you are right. If more disks are specified using `--resources` flag, then Mesos can send an offer with two or more disks.

Reply:
Could this code be run against a launch attempt for a pod with persistent volumes? Sorry, probably lacking some context here, but I don't fully see how persistent volumes aren't potentially a factor here.

Reply:
Persistent volume resources are passed separately.
```
    volumeMatchOption: Option[PersistentVolumeMatcher.VolumeMatch]
```